travis/clone_oca_dependencies

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/python
+"""Usage: clone_oca_dependencies [<checkout_dir> <build_dir>]
+
+Arguments:
+
+checkout_dir: the directory in which the dependency repositories will be cloned
+build_dir: the directory in which the tested repositories have been cloned
+
+If no arguments are provided, default to the layout used in the OCA travis
+configuration.
+
+The program will process the file oca_dependencies.txt at the root of the
+tested repository, and clone the dependency repositories in checkout_dir,
+before recursively processing the oca_dependencies.txt files of the
+dependencies.
+
+The expected format for oca_dependencies.txt:
+
+* comment lines start with # and are ignored
+* a dependency line contains:
+  - the name of the OCA project
+  - (optional) the URL to the git repository (defaulting to the OCA repository)
+  - (optional) the name of the branch to use (defaulting to ${VERSION})
+"""
+from __future__ import print_function
+import sys
+import os
+import os.path as osp
+import subprocess
+import logging
+
+
+_logger = logging.getLogger()
+
+
+def parse_depfile(depfile):
+    deps = []
+    for line in depfile:
+        line = line.strip()
+        if not line or line.startswith('#'):
+            continue
+        parts = line.split()
+        repo = parts[0]
+        if len(parts) > 2:
+            branch = parts[2]
+        else:
+            branch = os.environ.get('VERSION', '8.0')
+        if len(parts) > 1:
+            url = parts[1]
+        else:
+            url = 'https://github.com/OCA/%s' % repo
+        deps.append((repo, url, branch))
+    return deps
+
+
+def git_checkout(home, reponame, url, branch):
+    checkout_dir = osp.join(home, reponame)
+    if not osp.isdir(checkout_dir):
+        command = ['git', 'clone', url, '-b', branch, checkout_dir]
+        _logger.info('Calling %s', ' '.join(command))
+        subprocess.check_call(command)
+    return checkout_dir
+
+
+def run(home, build_dir):
+    dependencies = []
+    processed = set()
+    for repo in os.listdir(build_dir):
+        _logger.info('examining %s', repo)
+        processed.add(repo)
+        depfilename = osp.join(build_dir, repo, 'oca_dependencies.txt')
+        dependencies.append(depfilename)
+    for depfilename in dependencies:
+        try:
+            with open(depfilename) as depfile:
+                deps = parse_depfile(depfile)
+        except IOError:
+            deps = []
+        for depname, url, branch in deps:
+            _logger.info('* processing %s',  depname)
+            if depname in processed:
+                continue
+            processed.add(depname)
+            checkout_dir = git_checkout(home, depname, url, branch)
+            new_dep_filename = osp.join(checkout_dir, 'oca_dependencies.txt')
+            if new_dep_filename not in dependencies:
+                dependencies.append(new_dep_filename)
```

Comment:
I disagree. 

This will reduce breakage when other repositories' external dependencies change. And I personnally hate having to manually track these when a computer can perfectly do the job in my stead (and should do so).

@pedrobaeza Built time increase argument is void. The resulting clones are currently exactly the same as the one we are manually configuring, and the checkout time of OCA repositories is negligible compared to creating the base database.